internal/pkg/cli/app_show.go

```diff
@@ -139,23 +146,35 @@ func (opts *ShowAppOpts) retrieveData() error {
 	}
 
 	var serializedEnvs []serializedLoadBalancedWebAppEnv
+	var serializedServices []serializedLoadBalancedWebAppService
 	for _, env := range environments {
 		webAppURI, err := opts.identifier.URI(env.Name)
-		if err == nil {
-			serializedEnvs = append(serializedEnvs, serializedLoadBalancedWebAppEnv{
-				Name:   env.Name,
-				Region: env.Region,
-				Prod:   env.Prod,
-				URL:    webAppURI.DNSName,
-				Path:   webAppURI.Path,
-			})
+		if err != nil && !applicationNotDeployed(err) {
+			return fmt.Errorf("retrieving application URI: %w", err)
+		}
+		if err != nil && applicationNotDeployed(err) {
 			continue
 		}
```

Comment:
Should we check if the application is deployed before doing anything else to short circuit any logic in the case an application is not deployed to the env in context?